### PR TITLE
Migrate framework tests for rendering, semantics, widgets to null safety

### PR DESCRIPTION
**Pull request by darrenaustin**
## Description

Another block of framework tests migrated to NNDB.

### CODE REVIEW

File: packages/flutter/test/rendering/viewport_test.dart [hunk 1]
```diff
@@ -1705,15 +1703,15 @@ void main() {
     final RenderViewport renderViewport = RenderViewport(
       crossAxisDirection: AxisDirection.right, offset: ViewportOffset.zero()
     );
-    FlutterError error;
+    FlutterError? error;
```
Comment (goderbauer) [inline packages/flutter/test/rendering/viewport_test.dart:1706]: this could be late

File: packages/flutter/test/semantics/semantics_test.dart [hunk 2]
```diff
@@ -90,7 +87,7 @@ void main() {
       );
 
       {
-        FlutterError error;
+        FlutterError? error;
```
Comment (goderbauer) [inline packages/flutter/test/semantics/semantics_test.dart:90]: could be late

File: packages/flutter/test/semantics/semantics_test.dart [hunk 3]
```diff
@@ -107,13 +104,13 @@ void main() {
...
         expect(
-          error.diagnostics.singleWhere((DiagnosticsNode node) => node.level == DiagnosticLevel.hint).toString(),
+          error!.diagnostics.singleWhere((DiagnosticsNode node) => node.level == DiagnosticLevel.hint).toString(),
           'Instead of mutating the existing list, create a new list containing the desired `SemanticsNode`s.',
         );
       }
 
       {
-        FlutterError error;
+        FlutterError? error;
```
Comment (goderbauer) [inline packages/flutter/test/semantics/semantics_test.dart:113]: could be late

File: packages/flutter/test/widgets/animated_list_test.dart [hunk 4]
```diff
@@ -327,14 +325,14 @@ void main() {
     (WidgetTester tester) async {
     final GlobalKey key = GlobalKey();
     await tester.pumpWidget(Container(key: key));
-    FlutterError error;
+    FlutterError? error;
```
Comment (goderbauer) [inline packages/flutter/test/widgets/animated_list_test.dart:328]: could be late

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/test/rendering/viewport_test.dart (hunk 1 above)
Comment (darrenaustin) [reply packages/flutter/test/rendering/viewport_test.dart:1706]: Good point. I saw the `isNotNull` check expect below and thought there might be cases where it was null. But you are right, there is never a check where it is meant to be null. I'll fix this and the other. Thx.

**Review by gspencergoog — APPROVED** [review]

**Pull request merged**